types/instance.d.ts

```diff
@@ -113,39 +113,42 @@ type DecorationMethod<This, Return = This> = {
   (property: string | symbol, value: null|undefined, dependencies: string[]): Return;
 }
 
+export type FastifyDecorators = { fastify?: object, request?: object, reply?: object };
+
 /**
  * Fastify server instance. Returned by the core `fastify()` method.
  */
-export interface FastifyInstance<
+export type FastifyInstance<
```

Comment:
This is a breaking change since all of the code doing [declaration merging](https://fastify.dev/docs/latest/Reference/TypeScript/) (as supported and documented in version `5.x` of `fastify`) will no longer work. We should not convert `FastifyInstance` to a `type`.